test/testdata/lsp/code_actions/extract_class_func.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Extract method to module
+    'Hello'
+  end
+  
+  sig do
+    params(x: String)
+    .returns(String)
+  end
+  def name(x)
+    "#{Foo.greeting} #{x}"
+  end
+end
+
```

Comment:
What do you think about getting some non-standard method defs in place?

For example

```
class A
  prop :foo, Integer
end
```

This will implicitly define two methods (`foo` and `foo=`) and they'll both have a declLoc of the whole line.

Reply:
Moving props doesn't work with this implementation. I didn't dig too much, but my guess would be it doesn't work because it can't find the method body. That's a great catch! However, I'd put it as a stretch goal, because it's mostly related to the instance methods and ruby-infra is mostly interested in class methods. I'll add test to fix current behavior

Reply:
stretch goals are totally fine, all I was asking for in most of these comments was that we add tests to capture the current behavior, even if it doesn't yet do what we'd like it do to.